development/prow-installer/cmd/installer/main.go

```diff
@@ -87,13 +94,35 @@ func main() {
 	iamClient := serviceaccount.NewClient(readConfig.Prefix, iamService)
 	crmClient, err := roles.New(crmService)
 
-	for _, serviceAccount := range readConfig.ServiceAccounts {
+	for i, serviceAccount := range readConfig.ServiceAccounts {
 		// TODO implement handling error when SA already exists in GCP
-		if _, err := iamClient.CreateSA(serviceAccount.Name, readConfig.Project); err != nil {
+		if sa, err := iamClient.CreateSA(serviceAccount.Name, readConfig.Project); err != nil {
 			log.Errorf("Error creating Service Account %v", err)
 		} else {
+			key, err := iamClient.CreateSAKey(sa.Email)
+			if err != nil {log.Errorf("failed create serviceaccount %s key, got: %w", sa.Name, err)}
 			//log.Println(iamClient.CreateSAKey(sa.Email))
+			readConfig.ServiceAccounts[i].Key = key
```

Comment:
Is accessing a field using index `i` really needed since `serviceAccount` variable defined in for loop should already hold the same field?

Reply:
As far I know for range return copy of element. In this case I'm assigning key back in the readConfig struct.